tune_sklearn/tune_basesearch.py

```diff
@@ -194,51 +194,52 @@ def _check_is_fitted(self, method_name):
             self.max_iters = max_iters
-            if isinstance(scheduler, str):
-                if scheduler in TuneBaseSearchCV.defined_schedulers:
-                    if scheduler == "PopulationBasedTraining":
-                        self.scheduler = PopulationBasedTraining(
+            if isinstance(early_stopping, str):
+                if early_stopping in TuneBaseSearchCV.defined_schedulers:
+                    if early_stopping == "PopulationBasedTraining":
+                        self.early_stopping = PopulationBasedTraining(
                             metric="average_test_score")
-                    elif scheduler == "AsyncHyperBandScheduler":
-                        self.scheduler = AsyncHyperBandScheduler(
+                    elif early_stopping == "AsyncHyperBandScheduler":
+                        self.early_stopping = AsyncHyperBandScheduler(
                             metric="average_test_score")
-                    elif scheduler == "HyperBandScheduler":
-                        self.scheduler = HyperBandScheduler(
+                    elif early_stopping == "HyperBandScheduler":
+                        self.early_stopping = HyperBandScheduler(
                             metric="average_test_score")
-                    elif scheduler == "HyperBandForBOHB":
-                        self.scheduler = HyperBandForBOHB(
+                    elif early_stopping == "HyperBandForBOHB":
+                        self.early_stopping = HyperBandForBOHB(
                             metric="average_test_score")
-                    elif scheduler == "MedianStoppingRule":
-                        self.scheduler = MedianStoppingRule(
+                    elif early_stopping == "MedianStoppingRule":
+                        self.early_stopping = MedianStoppingRule(
                             metric="average_test_score")
-                    elif scheduler == "ASHAScheduler":
-                        self.scheduler = ASHAScheduler(
+                    elif early_stopping == "ASHAScheduler":
+                        self.early_stopping = ASHAScheduler(
                             metric="average_test_score")
                 else:
                     raise ValueError("{} is not a defined scheduler. "
                                      "Check the list of available schedulers."
-                                     .format(scheduler))
-            elif isinstance(scheduler, TrialScheduler):
-                self.scheduler = scheduler
-                self.scheduler.metric = "average_test_score"
-            elif scheduler is None:
-                self.scheduler = ASHAScheduler(metric="average_test_score")
+                                     .format(early_stopping))
+            elif isinstance(early_stopping, TrialScheduler):
+                self.early_stopping = early_stopping
+                self.early_stopping.metric = "average_test_score"
+            elif early_stopping is None:
```

Comment:
This case can never be hit because `early_stopping` cannot be None here. I think this case is covered by the outer else clause.

Reply:
Good catch thanks